File: nan21.dnet.module.md.ui.extjs/src/main/resources/webapp/js/ds/FiscalPeriod_Ds.js
/**
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
	
Ext.define(Dnet.ns.md + "FiscalPeriod_Ds", {
	extend: 'Ext.data.Model',
	
	validations: [
		{field: "fiscalYear", type: 'presence'},
		{field: "startDate", type: 'presence'},
		{field: "endDate", type: 'presence'},
		{field: "type", type: 'presence'},
		{field: "name", type: 'presence'}
	],
	
	fields: [
		{name:"type", type:"string"},
		{name:"startDate", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"endDate", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"posting", type:"boolean"},
		{name:"fiscalYearId", type:"string"},
		{name:"fiscalYear", type:"string"},
		{name:"id", type:"string"},
		{name:"name", type:"string"},
		{name:"description", type:"string"},
		{name:"notes", type:"string"},
		{name:"active", type:"boolean"},
		{name:"clientId", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"version", type:"int", useNull:true},
		{name:"refid", type:"string"},
		{name:"entityAlias", type:"string"},
		{name:"entityFqn", type:"string"}
	]
});

File: nan21.dnet.module.md.presenter/src/main/java/net/nan21/dnet/module/md/presenter/impl/base/model/FiscalPeriod_Ds.java
/**
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.md.presenter.impl.base.model;

import java.util.Date;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.Param;
import net.nan21.dnet.core.api.annotation.RefLookup;
import net.nan21.dnet.core.api.annotation.RefLookups;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.AbstractTypeDs;
import net.nan21.dnet.module.md.domain.impl.base.FiscalPeriod;
import net.nan21.dnet.module.md.domain.impl.base.FiscalYear;

@Ds(entity = FiscalPeriod.class, sort = {@SortField(field = FiscalPeriod_Ds.f_startDate, desc = true)})
@RefLookups({@RefLookup(refId = FiscalPeriod_Ds.f_fiscalYearId, namedQuery = FiscalYear.NQ_FIND_BY_NAME, params = {@Param(name = "name", field = FiscalPeriod_Ds.f_fiscalYear)})})
public class FiscalPeriod_Ds extends AbstractTypeDs<FiscalPeriod> {

	public static final String f_type = "type";
	public static final String f_startDate = "startDate";
	public static final String f_endDate = "endDate";
	public static final String f_posting = "posting";
	public static final String f_fiscalYearId = "fiscalYearId";
	public static final String f_fiscalYear = "fiscalYear";

	@DsField
	private String type;

	@DsField
	private Date startDate;

	@DsField
	private Date endDate;

	@DsField
	private Boolean posting;

	@DsField(join = "left", path = "fiscalYear.id")
	private String fiscalYearId;

	@DsField(join = "left", path = "fiscalYear.name")
	private String fiscalYear;

	public FiscalPeriod_Ds() {
		super();
	}

	public FiscalPeriod_Ds(FiscalPeriod e) {
		super(e);
	}

	public String getType() {
		return this.type;
	}

	public void setType(String type) {
		this.type = type;
	}

	public Date getStartDate() {
		return this.startDate;
	}

	public void setStartDate(Date startDate) {
		this.startDate = startDate;
	}

	public Date getEndDate() {
		return this.endDate;
	}

	public void setEndDate(Date endDate) {
		this.endDate = endDate;
	}

	public Boolean getPosting() {
		return this.posting;
	}

	public void setPosting(Boolean posting) {
		this.posting = posting;
	}

	public String getFiscalYearId() {
		return this.fiscalYearId;
	}

	public void setFiscalYearId(String fiscalYearId) {
		this.fiscalYearId = fiscalYearId;
	}

	public String getFiscalYear() {
		return this.fiscalYear;
	}

	public void setFiscalYear(String fiscalYear) {
		this.fiscalYear = fiscalYear;
	}
}
